server/src/main/java/uk/ac/cam/echo/extrusionfinder/preprocessor/ProfileDetector.java

```diff
@@ -0,0 +1,318 @@
+package uk.ac.cam.echo.extrusionfinder.preprocessor;
+
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfInt;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.highgui.Highgui;
+import org.opencv.imgproc.Imgproc;
+import java.util.List;
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.io.IOException;
+
+import uk.ac.cam.echo.extrusionfinder.imagedata.RGBImageData;
+import uk.ac.cam.echo.extrusionfinder.imagedata.GrayscaleImageData;
+
+/**
+ * Extracts extrusion cross-section profile.
+ */
+public class ProfileDetector {
+    private static final int imageSize = 200;
+
+    private int diameter;
+
+    /**
+     * Creates a profile detector.
+     * <p>
+     * The detector can be reused for different input images.
+     */
+    public ProfileDetector() {
+        // Ensure library is loaded.
+        System.loadLibrary("opencv_java249");
+
+        // If needed, allocate stuff for in-place operations.
+    }
+
+    /**
+     * Produces a grey scale image of only the extrusion's profile.
+     *
+     * @param bufferedImageIn An input photograph or scan containing extrusion.
+     * @return A newly allocated greyscale image of the extrusion profile.
+     */
+    public GrayscaleImageData process(RGBImageData input) {
+        diameter = imageSize;
```

Comment:
I think passing diameter into functions rather than setting a class variable is less yucky